EZP-27712: As a Developer I want Search to handle RemoveTranslationSignal

alongosz opened the pull request:
Implements [EZP-27712](https://jira.ez.no/browse/EZP-27712)
----

This PR adds search slot reindexing content on `RemoveTranslationSignal` added in #2031

**TODO**:
- [x] Add search engine indexing integration test checking search index after performing `ContentService::removeTranslation`
- [x] Create Search Slot for `RemoveTranslationSignal`
- [x] Enable created Slot for SQL Search
- [x] Enable created Slot for Elasticsearch
- [x] Enable created Slot for Solr (separate Solr Bundle PR ezsystems/ezplatform-solr-search-engine#98)
- [x] Remove TMP commit before merge

adamwojs commented on eZ/Publish/API/Repository/Tests/SearchEngineIndexingTest.php:
```diff
@@ -1154,6 +1203,46 @@ protected function createContentWithNameAndDescription($contentName, $contentDes
+    {
+        $repository = $this->getRepository();
+        $contentService = $repository->getContentService();
+        $locationService = $this->getRepository()->getLocationService();
```
You have a reference to repository already, so you don't need to call again  `$this->getRepository()`

adamwojs commented on eZ/Publish/API/Repository/Tests/SearchEngineIndexingTest.php:
```diff
@@ -1154,6 +1203,46 @@ protected function createContentWithNameAndDescription($contentName, $contentDes
 
     /**
+     * Create and publish a content with specified, in multiple languages, fields.
+     * $parentLocationIdList.
```
> $parentLocationIdList.

Judging by the dot at the end of previous line that this could be removed ?

alongosz commented:
> It looks good for me, besides of two nitpick comments

Good catches, thanks @adamwojs 
Fixed and squashed fixup. Also removed TMP commit, so now test on Solr fails.
Ref. passing build: https://travis-ci.org/ezsystems/ezpublish-kernel/builds/259804804

alongosz commented on eZ/Publish/API/Repository/Tests/SearchEngineIndexingTest.php:
```diff
@@ -1052,6 +1052,55 @@ public function testAssignSection()
+            // Test Location Search returns Content without removed Translation
+            $query = new LocationQuery(
+                [
+                    'query' => new Criterion\FullText('BrE'),
```
Side question for @andrerom and @pspanja:
I had to wrap this in `if`, because apparently our ES implementation does not have Location FullText Criterion visitor (there's one for Content). Is it one of the things *todo* for ES or is it that this is just not possible for ES? (For Solr we have two separate FullText Criterion visitors).

If it's the latter case, then maybe [ES dispatcher](https://github.com/ezsystems/ezpublish-kernel/blob/v6.10.1/eZ/Publish/Core/Search/Elasticsearch/Content/CriterionVisitorDispatcher.php#L94-L96) should not throw exception? (it really doesn't solve the need for this if, but I still think it would be better - we also did something similar for Facets AFAIR)

andrerom replied:
Unsure without digging into it myself.

Additional question would be if @MarioBlazek has gotten started on looking into improving elastic bundle, and if so helping him getting first set of improvements in might help on this as well in a way that is win win.

andrerom replied:
> If it's the latter case, then maybe ES dispatcher should not throw exception? 

won't be the same as with facets, User won't have any hint of why his criterion is not taken into account in this case. So -1, we stick with this as is, and cleanup once ES matures.

andrerom commented:
Note to self: Merge after Solr PR has been merged to 1.4 and master.
